library/swift/src/Configuration.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+
+/// Error that may be thrown by the configuration type.
+@objc
+public enum ConfigurationError: Int, Swift.Error {
+    /// Not all keys within the provided template were resolved.
+    case unresolvedTemplateKey
+}
+
+/// Builder used for creating configurations for new Envoy instances.
+@objcMembers
+public final class Configuration: NSObject {
+    /// The address at which Envoy should listen for requests.
+    public private(set) var address: String = "0.0.0.0"
+
+    /// The port at which Envoy should listen for requests.
+    public private(set) var port: String = "9001"
```

Comment:
I'd suggest dropping `address` and `port` since default configuration will be targeted at the direct API, which won't have a socket-based listener.

Reply:
Fine with me